firefox-ios/firefox-ios-tests/Tests/ClientTests/Coordinators/HomepageCoordinatorTests.swift

```diff
@@ -0,0 +1,83 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+import ComponentLibrary
+@testable import Client
+
+class HomepageCoordinatorTests: XCTestCase {
+    var profile: MockProfile!
+    var wallpaperManager: WallpaperManagerMock!
+    var router: MockRouter!
+
+    override func setUp() {
+        super.setUp()
+        profile = MockProfile()
+        wallpaperManager = WallpaperManagerMock()
+        router = MockRouter(navigationController: MockNavigationController())
+        DependencyHelperMock().bootstrapDependencies()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        DependencyHelperMock().reset()
+        profile = nil
+        wallpaperManager = nil
+        router = nil
+    }
+
+    func test_showWallpaperSelectionOnboarding_withNoZeroSearch_doesNotPresentWallpaperSelection() {
+        let coordinator = HomepageCoordinator(
+            windowUUID: .XCTestDefaultUUID,
+            profile: profile,
+            wallpaperManager: wallpaperManager,
+            isZeroSearch: false,
+            router: router
+        )
+        trackForMemoryLeaks(coordinator)
```

Comment:
```suggestion
    let coordinator = createSubject(isZeroSearch: false)
```

nit; I personally like having this code in a `createSubject` function to avoid repetition. IMO I think it helps readability of the test, to know what is the Given-when-then parts of it

Reply:
I am happy to make this change. I thought the duplication made it more clear what was happening since `createSubject` felt a bit opaque to me but I can move this into a shared function!